bolt-modules/boltlib/lib/puppet/functions/run_task.rb

```diff
@@ -77,7 +77,7 @@ def run_task_with_description(task_name, targets, description, args = {})
     options[:description] = description if description
 
     # Don't bother loading the local task definition if all targets use the 'pcp' transport.
-    if !targets.empty? && targets.all? { |t| t.transport == 'pcp' }
+    if (pcp_only = targets.any? && targets.all? { |t| t.transport == 'pcp' })
```

Comment:
I wish there was a more idomatic way to do this in ruby...

Reply:
What if we move the `if targets.empty?` check up near the top of the function. All this intermediate logic is superfluous in that case anyway so it makes sense for it to be an early test.